src/event_loop.rs

```diff
@@ -370,12 +391,82 @@ impl<T> EventLoopWindowTarget<T> {
         #[cfg(any(x11_platform, wasm_platform, wayland_platform, windows))]
         self.p.listen_device_events(_allowed);
     }
+
+    /// Get an [`OwnedDisplayHandle`] that keeps the underlying display handle alive.
+    ///
+    /// This is necessary to use the display handle outside of the event loop, for example for
+    /// OpenGL setup.
+    pub fn owned_display_handle(&self) -> OwnedDisplayHandle {
+        OwnedDisplayHandle {
+            p: self.p.owned_display_handle(),
+        }
+    }
 }
 
 unsafe impl<T> HasRawDisplayHandle for EventLoopWindowTarget<T> {
     /// Returns a [`raw_window_handle::RawDisplayHandle`] for the event loop.
-    fn raw_display_handle(&self) -> RawDisplayHandle {
-        self.p.raw_display_handle()
+    fn raw_display_handle(&self) -> Result<RawDisplayHandle, HandleError> {
+        Ok(self.p.raw_display_handle())
+    }
+}
+
+impl<T> HasDisplayHandle for EventLoopWindowTarget<T> {
+    fn display_handle(&self) -> Result<DisplayHandle<'_>, HandleError> {
+        // SAFETY: The returned display handle is always valid for this lifetime.
+        self.raw_display_handle()
+            .map(|handle| unsafe { DisplayHandle::borrow_raw(handle) })
+    }
+}
+
+unsafe impl<T> raw_window_handle_05::HasRawDisplayHandle for EventLoopWindowTarget<T> {
+    fn raw_display_handle(&self) -> raw_window_handle_05::RawDisplayHandle {
+        let result = HasRawDisplayHandle::raw_display_handle(self);
+
+        match result {
+            Ok(handle) => cvt_rdh_06_to_05(handle),
+            Err(e) => panic!("Failed to get raw display handle: {:?}", e),
+        }
+    }
+}
+
+/// An owned handle used to represent the underlying display.
+///
+/// It is sometimes necessary to pass a display handle to a function without involving any
+/// windows. This comes in play, for example, for OpenGL setup. However, it is impossible to hold
+/// a reliable reference to the [`EventLoop`] or an [`EventLoopWindowTarget`] while the event loop
+/// is running. This type aims to fill that gap by being a generic-free, cheaply clonable handle
+/// representing the display server.
+///
+/// If you already have a [`Window`], you can get a display handle from it instead of this.
+///
+/// [`Window`]: crate::window::Window
+#[derive(Clone)]
+pub struct OwnedDisplayHandle {
```

Comment:
I'm uncertain if this is the best design, and if it would maybe make sense to have this in the `raw-window-handle` crate instead? Though I guess the `Drop` impl can't really be done in that crate, so probably not :/

Reply:
It's unfortunately necessary for tying the lifetimes of things that need the raw display (like OpenGL) when there isn't a window available yet. As the `EventLoop` must be held exclusively to be polled, we need another type to represent the actual underlying display that can be passed to OpenGL/wgpu/etc.

"Display" is used here as a term for the connector between the client application and the window manager server, like `xcb_connection` or `wl_display`. I'm open to better names.

Reply:
Undestood - I've asked follow-up questions on Matrix